### PR TITLE
 Configures integration testing project with env var and uses different images for each test.

### LINKED ISSUE #274: BuildImage/BuildDocker/BuildTar integration tests use the same image names for each run, and doesn't remove the test images between runs
An integration test could fails to push the image, but succeed to run the image pushed from a previous integration test, which would result in the test passing.

### LINKED ISSUE #588: Make integration testing GCP project configurable.
See https://github.com/GoogleContainerTools/jib/pull/587#issuecomment-404552430

This should help contributors with running integration tests.

### PR DESCRIPTION
**Pull request by coollog**
Fixes #588
Fixes #274

Set the env var `JIB_INTEGRATION_TESTING_PROJECT` to `jib-integration-testing`, for example.

### CODE REVIEW

File: jib-gradle-plugin/src/integration-test/java/com/google/cloud/tools/jib/gradle/JibPluginIntegrationTest.java [hunk 1]
```diff
@@ -94,28 +99,47 @@ private static String buildToDockerDaemonAndRun(TestProject testProject, String
...
-                "{{.Created}}",
-                "gcr.io/jib-integration-testing/emptyimage:gradle")
-            .run()
-            .trim());
+        new Command("docker", "inspect", "-f", "{{.Created}}", targetImage).run().trim());
   }
 
   @Test
   public void testBuild_simple() throws IOException, InterruptedException {
+    String targetImage = "gcr.io/jib-integration-testing/simpleimage:gradle" + System.nanoTime();
```
Comment (TadCordle) [inline jib-gradle-plugin/src/integration-test/java/com/google/cloud/tools/jib/gradle/JibPluginIntegrationTest.java]: Should this be using the environment variable?

File: jib-core/src/main/java/com/google/cloud/tools/jib/image/ReproducibleLayerBuilder.java [hunk 2]
```diff
@@ -44,6 +45,13 @@
       throws IOException {
     List<TarArchiveEntry> tarArchiveEntries = new ArrayList<>();
 
+    // Adds the extraction path itself; to do this, we are adding the current directory to act as
+    // the file input (since all directories are treated the same in TarArchiveEntry).
```
[the remark this answers is not in the document]
Comment (coollog) [reply jib-core/src/main/java/com/google/cloud/tools/jib/image/ReproducibleLayerBuilder.java]: Oh oops.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -42,7 +42,7 @@ To build, use the provided `build.sh` which builds and tests each of the compone
 5. Associate the change with an existing issue or file a [new issue](../../issues).
 6. Create a pull request!
 
-**Note** that you will not be able to run the integration tests for `jib-gradle-plugin` or `jib-maven-plugin` because those push to our integration-testing GCP project. If you would like to run integration tests, change all uses of the `jib-integration-testing` project to your own GCP project and run `./gradlew integrationTest` for `jib-gradle-plugin` and `./mvnw verify -Pintegration-tests` for `jib-maven-plugin`.
+**Note** that in order to run integration tests, you will need to set the environment variable `JIB_INTEGRATION_TESTING_PROJECT` to the GCP project you would like to use for testing. Otherwise, feel free to skip integration tests.
```
Comment (TadCordle) [inline CONTRIBUTING.md]: Maybe we should also mention that they'll need docker installed to run the integration tests.

**Comment by coollog** [comment]
@GoogleContainerTools/java-tools Make sure to add `export JIB_INTEGRATION_TESTING_PROJECT=jib-integration-testing` to your bashrc!

**Comment by TadCordle** [comment]
Ah, we should add that to the kokoro scripts as well I think

**Pull request merged**